lib/ClangImporter/ClangImporter.cpp

```diff
@@ -7471,6 +7471,37 @@ bool importer::isForeignReferenceTypeWithoutImmortalAttrs(const clang::QualType
          !hasImmortalAtts(pointeeType->getDecl());
 }
 
+// TODO: Instead of just checking the immediate parent, do a traversal till the
+// top most parent
+std::pair<bool, clang::CXXRecordDecl *>
+getFRTInfoFromBaseDecls(swift::ASTContext &ctx, Evaluator &evaluator,
+                        const clang::RecordDecl *decl) {
+  bool declHasAReferenceTypeBase = false;
+  clang::CXXRecordDecl *baseRefDecl = nullptr;
+
+  if (auto *cxxRecordDecl = llvm::dyn_cast<clang::CXXRecordDecl>(decl)) {
+    auto hasReferenceTypeBase =
+        [&evaluator, &ctx, &baseRefDecl](const clang::CXXBaseSpecifier &base) {
+          if (clang::CXXRecordDecl *baseDecl =
+                  base.getType()->getAsCXXRecordDecl()) {
+            auto semanticsKind = evaluateOrDefault(
+                evaluator, CxxRecordSemantics({baseDecl, ctx}), {});
+            if (semanticsKind == CxxRecordSemanticsKind::Reference) {
+              baseRefDecl = baseDecl;
+              return true;
+            }
+          }
+          return false;
+        };
+
+    declHasAReferenceTypeBase =
+        std::any_of(cxxRecordDecl->bases().begin(),
+                    cxxRecordDecl->bases().end(), hasReferenceTypeBase);
```

Comment:
Instead of `std::any_of`, you could use `llvm::any_of` which would accept a range `cxxRecordDecl->bases()` as a parameter.